includes/class-fs-logger.php

```diff
@@ -11,6 +11,10 @@
 	}
 
 	class FS_Logger {
+        /**
+         * @var bool
+         */
+        public static  $ON          = false;
```

Comment:
@DanieleAlessandra I think we are mixing two things here :)

We have the function `_set_storage_logging` which only deals with DB related storage. So making it `off` shouldn't really turn off regular logging.

## For calling `$wpdb->insert` when we have just turned off the logging

We need to update the already existing `$_isStorageLoggingOn` static variable inside the `_set_storage_logging`. Something like

```php
public static function _set_storage_logging( $is_on = true ) {
    // .. existing code

    self::$_isStorageLoggingOn = $is_on;
}
```

and that's it. I think this will work just fine.

## Out of sync DB creation issue

I see you've made adjustment in the creation statement with `CREATE TABLE IF NOT EXISTS`. I propose we do it like this:

1. Drop table if exists.
2. Then create the table.

I think this way when the logging is turned on we are 100% sure to get the table schema we are after. Since we are dropping table anyway when turning off, I think this approach is fine.

---

So kindly see if the above suggestion would work.